source/frontend/src/components/common/mapFSM/MapStateMachineContext.tsx

```diff
@@ -0,0 +1,275 @@
+    serviceSend({
+      type: 'PROCESS_FIT_BOUNDS',
+    });
+  }, [serviceSend]);
+
+  const closePopup = useCallback(() => {
+    serviceSend({ type: 'CLOSE_POPUP' });
+  }, [serviceSend]);
+
+  const refreshMapProperties = useCallback(() => {
+    serviceSend({ type: 'REFRESH_PROPERTIES' });
+  }, [serviceSend]);
+
+  const setMapFilter = useCallback(
+    (mapFilter: IPropertyFilter) => {
+      serviceSend({ type: 'SET_MAP_FILTER', mapFilter });
+    },
+    [serviceSend],
+  );
+
+  const prepareForCreation = useCallback(() => {
+    serviceSend({ type: 'PREPARE_FOR_CREATION' });
+  }, [serviceSend]);
+
+  const startSelection = useCallback(() => {
+    serviceSend({ type: 'START_SELECTION' });
+  }, [serviceSend]);
+
+  const finishSelection = useCallback(() => {
+    serviceSend({ type: 'FINISH_SELECTION' });
+  }, [serviceSend]);
+
+  const setDraftLocations = useCallback(
+    (locations: LatLngLiteral[]) => {
+      serviceSend({ type: 'SET_DRAFT_LOCATIONS', locations });
+    },
+    [serviceSend],
+  );
+
+  const showPopup = useMemo(() => {
+    return state.context.mapLocationFeatureDataset !== null;
+  }, [state.context.mapLocationFeatureDataset]);
+
+  return (
+    <MapStateMachineContext.Provider
+      value={{
+        isSidebarOpen: [
+          { mapVisible: { sideBar: 'sidebarOpen' } },
+          { mapVisible: { sideBar: 'selecting' } },
+        ].some(state.matches),
+        pendingFlyTo: state.matches({ mapVisible: { mapRequest: 'pendingFlyTo' } }),
+        requestedFlyTo: state.context.requestedFlyTo,
+        mapFeatureSelected: state.context.mapFeatureSelected,
+        mapLocationSelected: state.context.mapLocationSelected,
+        mapLocationFeatureDataset: state.context.mapLocationFeatureDataset,
+        selectedFeatureDataset: state.context.selectedFeatureDataset,
+        showPopup: showPopup,
+        isLoading: state.context.isLoading,
+        mapFilter: state.context.mapFilter,
+        mapFeatureData: state.context.mapFeatureData,
+        draftLocations: state.context.draftLocations,
+        pendingFitBounds: state.matches({ mapVisible: { mapRequest: 'pendingFitBounds' } }),
+        requestedFitBounds: state.context.requestedFitBounds,
+        iSelecting: state.matches({ mapVisible: { sideBar: 'selecting' } }),
+
+        setMapFilter,
+        refreshMapProperties,
+        processFlyTo,
+        processFitBounds,
+        openSidebar,
+        closeSidebar,
+        requestFlyToLocation,
+        requestFlyToBounds,
+        mapClick,
+        mapMarkerClick,
+        closePopup,
+        prepareForCreation,
+        startSelection,
+        finishSelection,
+        setDraftLocations,
+      }}
+    >
+      {children}
+    </MapStateMachineContext.Provider>
+  );
+};
+
+const getQueryParams = (filter: IPropertyFilter): IGeoSearchParams => {
+  // The map will search for either identifier.
+  const pinOrPidValue = filter.pinOrPid ? filter.pinOrPid?.replace(/-/g, '') : undefined;
```

Comment:
this looks like a regression on the previously implemented request to accept spacse as well as hyphens as pid delimiters.

Reply:
Right now is the same method that is being used in dev. https://github.com/bcgov/PSP/blob/335915188e90ec530cfb91749a9ecc5b0d8bfefa/source/frontend/src/components/maps/MapView.tsx#L44

The loadPropertiesRequest does have the extra regex parameters as in dev.